Zotero/Scenes/Master/Collections/Views/ExpandableCollectionsCollectionViewHandler.swift

```diff
@@ -68,122 +86,48 @@ final class ExpandableCollectionsCollectionViewHandler: NSObject {
 
         if !snapshot.visibleItems.contains(where: { $0.identifier == collectionId }) {
             // Selection is collapsed, we need to expand and select it then
-            self.update(with: tree, selectedId: collectionId, animated: false) { [weak self] in
-                self?.selectIfNeeded(collectionId: collectionId, tree: tree, scrollToPosition: scrollToPosition)
+            update(with: tree, selectedId: collectionId, animated: false) { [weak self] in
+                inMainThread { [weak self] in
+                    self?.selectIfNeeded(collectionId: collectionId, tree: tree, scrollToPosition: scrollToPosition)
+                }
```

Comment:
This completion block seems to only be passed to datasource.apply as well, so it can be simplified.
```suggestion
                self?.selectIfNeeded(collectionId: collectionId, tree: tree, scrollToPosition: scrollToPosition)
```

Reply:
Yeah good catch, 5089eda22716b95ad850688ffe201584881162c2